dora/src/masm/x64.rs

```diff
@@ -474,6 +474,44 @@ impl MacroAssembler {
         }
     }
 
+    pub fn int_rol(&mut self, mode: MachineMode, dest: Reg, lhs: Reg, rhs: Reg) {
+        let x64 = match mode {
+            MachineMode::Int32 => 0,
+            MachineMode::Int64 => 1,
+            _ => unimplemented!(),
+        };
+
```

Comment:
Shall we assert here that `lhs` should not be `RCX`?

Reply:
Yes, good point!

Reply:
Mhhh, aren't we doing this two lines below?

Reply:
Oh right, missed that.... Sorry...

Reply:
No problem! So this is good to go, right?